integration/xgboost/processor/src/nvflare-plugin/local_mock.cc

```diff
@@ -0,0 +1,128 @@
+/**
+ * Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <cstring>
+#include "local_mock.h"
+#include "data_set_ids.h"
+
+void* LocalMockProcessor::ProcessHistograms(std::size_t *size, const std::vector<double>& histograms) {
+    if (debug_) {
+        std::cout << "ProcessHistograms called with " << histograms.size() << " entries" << std::endl;
+    }
+    DamEncoder encoder(kDataSetHistogramResult, true);
+    encoder.AddFloatArray(histograms);
+    return encoder.Finish(*size);
```

Comment:
This func returns a dynamically allocated buffer, who is responsible for freeing it?

Reply:
This is freed in ProcessGHPairs().